services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/logging/DisclosureLoggingAspect.kt

```diff
@@ -0,0 +1,116 @@
+package fi.hel.haitaton.hanke.logging
+
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentMetadataDto
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentMetadataDto
+import fi.hel.haitaton.hanke.banners.BannerResponse
+import fi.hel.haitaton.hanke.currentUserId
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.domain.PublicHanke
+import fi.hel.haitaton.hanke.gdpr.CollectionNode
+import fi.hel.haitaton.hanke.hakemus.HakemusDeletionResultDto
+import fi.hel.haitaton.hanke.hakemus.HakemusResponse
+import fi.hel.haitaton.hanke.hakemus.HakemusWithExtrasResponse
+import fi.hel.haitaton.hanke.hakemus.HankkeenHakemuksetResponse
+import fi.hel.haitaton.hanke.permissions.HankeKayttajaController
+import fi.hel.haitaton.hanke.permissions.HankeKayttajaDto
+import fi.hel.haitaton.hanke.permissions.HankeKayttajaResponse
+import fi.hel.haitaton.hanke.permissions.HankekayttajaDeleteService
+import fi.hel.haitaton.hanke.permissions.WhoamiResponse
+import fi.hel.haitaton.hanke.profiili.Names
+import fi.hel.haitaton.hanke.taydennys.TaydennysResponse
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulos
+import org.aspectj.lang.annotation.AfterReturning
+import org.aspectj.lang.annotation.Aspect
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Component
+
+@Aspect
+@Component
+class DisclosureLoggingAspect(private val disclosureLogService: DisclosureLogService) {
+    @AfterReturning(
+        pointcut = "@annotation(io.swagger.v3.oas.annotations.Operation)", returning = "result")
+    fun logResponse(result: Any?) {
```

Comment:
Maybe this function could be unit-tested?

Reply:
I added unit tests. I also added specific exceptions to replace the generic RuntimeExceptions.